console/src/schematic/hooks.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 Synnax Labs, Inc.
+//
+// Use of this software is governed by the Business Source License included in the file
+// licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with the Business Source
+// License, use of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt.
+
+import { UnexpectedError } from "@synnaxlabs/client";
+import { Status, Synnax } from "@synnaxlabs/pluto";
+import { useMutation } from "@tanstack/react-query";
+import { open, save } from "@tauri-apps/plugin-dialog";
+import { readFile, writeFile } from "@tauri-apps/plugin-fs";
+import { useDispatch, useStore } from "react-redux";
+
+import { Confirm } from "@/confirm";
+import { Layout } from "@/layout";
+import { select } from "@/schematic/selectors";
+import { fileHandler } from "@/schematic/services/file";
```

Comment:
This import causes a dependency flow in a direction it shouldn't go. The entire point of the `services` package is to  isolated it from the root `schematic` package. Importing this dependency links the two again, potentially causing issues with circular deps in the future.